cmd/parca-agent/main.go

```diff
@@ -390,17 +260,25 @@ func main() {
 		})
 	}
 
-	// Run group for target manager
+	// Add profiler.
+	profiler := pp.AddProfiler(ctx, profiler.NewCPUProfiler, func() map[int]model.LabelSet {
+		return m.ProcessLabels()
+	})
+
+	// Run group for profiler.
 	{
-		ctx, cancel := context.WithCancel(ctx)
 		g.Add(func() error {
-			level.Debug(logger).Log("msg", "starting target manager")
-			return tm.Run(ctx, m.SyncCh())
-		}, func(error) {
-			cancel()
+			return profiler.Run(ctx)
+		}, func(err error) {
+			profiler.Stop()
+			level.Error(logger).Log("msg", "profiler ended with", "error", err, "profilerName", profiler.Name())
 		})
 	}
 
+	if err != nil {
+		panic("fatal error: could not start profiler")
```

Comment:
```suggestion
		panic("fatal error: could not start profiler")
```
		level.Error(logger).Log("fatal error: could not start profiler", err)
 		os.Exit(1)

better than panicing

Reply:
Ohh this was a bad merge, good eye!!